### PR TITLE
Add number of waiting tasks by types in test file records response

### PR DESCRIPTION
**Pull request by shuofan**
#### What type of this PR
feature

#### What this PR does / why we need it:
Add number of waiting tasks by types in test file records response

#### Specified Reviewers:

/assign @Effet @sfwn

### CODE REVIEW

File: apistructs/file_records.go [hunk 1]
```diff
@@ -89,5 +89,10 @@ type GetTestFileRecordResponse struct {
 
 type ListTestFileRecordsResponse struct {
 	Header
-	Data []TestFileRecord
+	Data *ListTestFileRecordsResponseData
```
Comment (Effet) [inline apistructs/file_records.go:92]: struct changes will break api compatibility, how do you deal with it?

File: modules/dop/dao/file_records.go [hunk 2]
```diff
@@ -85,18 +85,28 @@ func (client *DBClient) GetRecord(id uint64) (*TestFileRecord, error) {
...
-	return res, nil
+
+	var count int
+	counter := make(map[string]int)
+	for _, t := range req.Types {
+		if err := client.Model(&TestFileRecord{}).Where("`type` = ? AND `state` = ?", t, apistructs.FileRecordStatePending).Count(&count).Error; err != nil {
+			return nil, nil, err
+		}
+		counter[string(t)] = count
+	}
```
Comment (Effet) [inline modules/dop/dao/file_records.go:107]: for loop to query db is dangerous, you should use native sql `group by`.

**Review by Effet — CHANGES_REQUESTED** [review]

File: modules/dop/dao/file_records.go (hunk 2 above)
Comment (shuofan) [reply modules/dop/dao/file_records.go]: done

File: apistructs/file_records.go (hunk 1 above)
Comment (shuofan) [reply apistructs/file_records.go:92]: It is not backward-compatible, as not designed well at the beginning. We may version it in the future. Updated API docs here: https://erda.cloud/erda/workBench/projects/387/apps/5880/apiDesign

Comment (Effet) [reply apistructs/file_records.go:92]: so must frontend change their code at same time?

**Review by Effet — APPROVED** [review]

**Pull request merged**